File: spring_backennd/src/main/java/com/example/spring_backennd/data/Product.java
package com.example.spring_backennd.data;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Data
@NoArgsConstructor@AllArgsConstructor@ToString
public class Product  implements Serializable {
     @Id
     @Column(name = "product_id", nullable = false)
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long product_id;

    private String name;
    private String description ;
    private double currentPrice;
    private boolean promotion;
    private boolean selected;
    private  boolean available ;
    private String photoName;
    @ManyToOne
    private Category category;

    public Long getId() {
     return product_id;
    }

    public void setId(Long id) {
       this.product_id = id;
    }
}

File: angularfrontend/src/app/Models/Product.ts
import { Category } from 'src/app/resData/model';

export interface Product {
  id: number;
  name: string;
  description?: any;
  currentPrice: number;
  promotion: boolean;
  selected: boolean;
  available: boolean;
  photoName: string;
  category: Category;
}

export interface Embedded {
  products: Product[];
}

export interface ProductResData {
  _embedded: Embedded;
}
